src/stopwatch.hpp

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019-2021 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+//
+// Simple stopwatch class is for performing runtime speed tests.
+// Can specify upon construction the time units (in seconds, milliseconds, or
+// nanoseconds). Stopwatch uses milliseconds by default. Stopwatch has two states:
+// either running or not. User can only capture times through Elapsed(), Lap() or
+// Stop() while Stopwatch is running. Laps capture the elapsed time between each Lap()
+// call (as well as Start()/Lap() and Lap()/Stop() intervals). Total keeps a
+// culumative total of all Start()/Stop() time intervals. User can't call Stop() on a
+// stopped watch or Start() on a running watch. Clear() removes all memory and stops
+// watch by default, but can optionally restart watch immediatedly.
+
+#ifndef SRC_STOPWATCH_HPP_
+#define SRC_STOPWATCH_HPP_
+
+// ** Doctest include must go first for all header tests to run.
+#include "doctest.h"
+// **
+
+#include <stdio.h>
+
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <thread>
+#include <vector>
+
+#include "sugar.hpp"
+
+class Stopwatch {
+ public:
+  using clock = std::chrono::high_resolution_clock;
+  using time_point = std::chrono::time_point<clock>;
+  using time_duration = std::chrono::duration<uint16_t, std::nano>;
+
+  enum TimeScale { no_scale, second_scale, millisecond_scale, nanosecond_scale };
```

Comment:
From the contributing document:

Use [scoped enums](https://abseil.io/tips/86) rather than C enums.